### PR TITLE
virtcontainer: add error return code

### PR DESCRIPTION
**Pull request by alicefr**
Add error return code to append functions.

Fixes: #2035

Signed-off-by: Alice Frosi <afrosi@de.ibm.com>

### CODE REVIEW

**Comment by alicefr** [comment]
PR based on this [comment](https://github.com/kata-containers/runtime/pull/2034#discussion_r321172843). /cc @jodh-intel

**Comment by jodh-intel** [comment]
A Travis is failing on a static analysis issue:

```
virtcontainers/qemu_ppc64le.go:151: File is not `gofmt`-ed with `-s` (gofmt)
        devices, err = q.appendBlockDevice(devices, drive)
        if err != nil {
                return nil, err
        }
The command ".ci/static-checks.sh" failed and exited with 1 during .
```

File: virtcontainers/qemu_arch_base.go [hunk 1]
```diff
@@ -517,15 +521,15 @@ func genericNetwork(endpoint Endpoint, vhost, nestedRun bool, index int) (govmmQ
 	return d, nil
 }
 
-func (q *qemuArchBase) appendNetwork(devices []govmmQemu.Device, endpoint Endpoint) []govmmQemu.Device {
+func (q *qemuArchBase) appendNetwork(devices []govmmQemu.Device, endpoint Endpoint) ([]govmmQemu.Device, error) {
 	d, err := genericNetwork(endpoint, q.vhost, q.nestedRun, q.networkIndex)
 	if err != nil {
 		virtLog.WithField("subsystem", "qemuArch").WithError(err).Error("Failed to append network")
```
Comment (devimc) [inline virtcontainers/qemu_arch_base.go]: I think this log is no more needed

File: virtcontainers/qemu_arch_base.go [hunk 2]
```diff
@@ -548,14 +552,14 @@ func genericBlockDevice(drive config.BlockDrive, nestedRun bool) (govmmQemu.Bloc
 	}, nil
 }
 
-func (q *qemuArchBase) appendBlockDevice(devices []govmmQemu.Device, drive config.BlockDrive) []govmmQemu.Device {
+func (q *qemuArchBase) appendBlockDevice(devices []govmmQemu.Device, drive config.BlockDrive) ([]govmmQemu.Device, error) {
 	d, err := genericBlockDevice(drive, q.nestedRun)
 	if err != nil {
 		virtLog.WithField("subsystem", "qemuArch").WithError(err).Error("Failed to append block device")
```
Comment (devimc) [inline virtcontainers/qemu_arch_base.go]: same here

**Pull request merged**